### PR TITLE
topology2: Add support for 4 ch DMIC capture

### PR DESCRIPTION
**Pull request by ranj063**
fixes https://github.com/thesofproject/sof/issues/5678

### CODE REVIEW

**Comment by fredoh9** [comment]
I am able to verify this on ADL-P + AIOC + Transducer. I used both nocodec mode and HDA mode.

File: tools/topology/topology2/cavs/platform/intel/dmic-generic.conf [hunk 1]
```diff
@@ -84,6 +97,8 @@ Object.PCM {
 			name $DMIC0_PCM_CAPS
 			# only 32-bit capture supported now
 			formats 'S32_LE'
+			channels_min $NUM_DMICS
```
Comment (bardliao) [inline tools/topology/topology2/cavs/platform/intel/dmic-generic.conf:100]: If we set channels_min = $NUM_DMICS, we can only with channel number equal to $NUM_DMICS. In other words, we can't test with -c 2 if we set $NUM_DMICS = 4

Comment (ranj063) [reply tools/topology/topology2/cavs/platform/intel/dmic-generic.conf:100]: @bardliao what happens with -c 3 then? I dont think we want to set the min/max channels here to be 2/4. If the number of DMIC's is 4, we want to allow only 4 ch capture. @plbossart please correct me if I am wrong

Comment (lgirdwood) [reply tools/topology/topology2/cavs/platform/intel/dmic-generic.conf:100]: This should be defined by the min supported by the DMIC driver or based on the other settings defined as part of this config.

Comment (plbossart) [reply tools/topology/topology2/cavs/platform/intel/dmic-generic.conf:100]: that's right @ranj063, when we have 4channels we provide all 4 channels to the host. In the case of Chromebooks, the UCM file knows what channels are relevant on what platform and extract the useful channels in userspace.
Also for HDaudio platforms, we share the number of dmics with userspace, so pulseaudio/pipewire will open the PCM device with the relevant number of channels.

We currently don't have the ability (or need) to change the channel map depending on that the user asks for.

Comment (plbossart) [reply tools/topology/topology2/cavs/platform/intel/dmic-generic.conf:100]: and btw when we have 1 or 3 mics, we currently round up.

````
sof-hda-generic\;sof-hda-generic-1ch\;-DCHANNELS=2\;
sof-hda-generic\;sof-hda-generic-2ch\;-DCHANNELS=2\;
sof-hda-generic\;sof-hda-generic-3ch\;-DCHANNELS=4\;
sof-hda-generic\;sof-hda-generic-4ch\;-DCHANNELS=4\;
````

**Review by lgirdwood — APPROVED** [review]
Rounding up fine by me, IIRC we do this for lots of ASoC codecs today.

**Comment by sys-pt1s** [comment]
Can one of the admins verify this patch?

**Review by bardliao — APPROVED** [review]

**Pull request merged**